runtime/vara/src/tests.rs

```diff
@@ -20,15 +20,345 @@ use super::*;
 use crate::Runtime;
 use frame_support::{dispatch::GetDispatchInfo, traits::StorageInstance};
 use frame_system::limits::WeightsPerClass;
-use gear_core::costs::LazyPagesCosts;
+use gear_core::{
+    costs::{CostOf, LazyPagesCosts},
+    pages::GearPagesAmount,
+};
 use pallet_gear::{InstructionWeights, MemoryWeights, SyscallWeights};
 use pallet_staking::WeightInfo as _;
-use runtime_common::weights::{
-    check_instructions_weights, check_lazy_pages_costs, check_pages_costs, check_syscall_weights,
-    PagesCosts,
-};
 use sp_runtime::AccountId32;
 
+const INSTRUCTIONS_SPREAD: u8 = 50;
+const SYSCALL_SPREAD: u8 = 10;
+const PAGES_SPREAD: u8 = 10;
+
+/// Structure to hold weight expectation
+struct WeightExpectation {
+    weight: u64,
+    expected: u64,
+    spread: u8,
+    name: &'static str,
+}
+
+impl WeightExpectation {
+    fn new(weight: u64, expected: u64, spread: u8, name: &'static str) -> Self {
+        Self {
+            weight,
+            expected,
+            spread,
+            name,
+        }
+    }
+
+    fn check(&self) -> Result<(), String> {
+        let left = self.expected - self.expected * self.spread as u64 / 100;
+        let right = self.expected + self.expected * self.spread as u64 / 100;
+
+        if left > self.weight || self.weight > right {
+            return Err(format!("Instruction [{}]. Weight is {} ps. Expected weight is {} ps. {}% spread interval: [{left} ps, {right} ps]", self.name, self.weight, self.expected, self.spread));
+        }
+
+        Ok(())
+    }
+}
+
+fn check_expectations(expectations: &[WeightExpectation]) -> Result<(), Vec<String>> {
+    let errors = expectations
+        .iter()
+        .filter_map(|expectation| {
+            if let Err(err) = expectation.check() {
+                Some(err)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<String>>();
+
+    if errors.is_empty() {
+        Ok(())
+    } else {
+        Err(errors)
+    }
+}
+
+/// Memory pages access costs.
+struct PagesCosts {
+    pub load_page_data: CostOf<GearPagesAmount>,
+    pub upload_page_data: CostOf<GearPagesAmount>,
+    pub mem_grow: CostOf<GearPagesAmount>,
+    pub mem_grow_per_page: CostOf<GearPagesAmount>,
+    pub parachain_read_heuristic: CostOf<GearPagesAmount>,
+}
+
+impl<T: pallet_gear::Config> From<MemoryWeights<T>> for PagesCosts {
```

Comment:
btw we already have `gear_core::costs::LazyPagesCosts`. mb reuse it here?
 https://github.com/gear-tech/gear/blob/73929e309d40aceec8da761b8b2313eab7ea41a4/pallets/gear/src/schedule.rs#L1304

Reply:
we could add the missing fields (from `MemoryWeights<T>`) to `gear_core::costs::LazyPagesCosts` and reuse that for both gtest and these checks.

Reply:
Done

Reply:
That way LazyPagesCosts become not actually costs of lazy pages. wdyt @grishasobol ?

Reply:
Do you think we need a more generic name for the structure? @breathx